### PR TITLE
Species blacklist refactor/defines in species vars

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
The magic mirror now uses the species blacklisted var instead of a list of strings

Used defines for various species vars

This also fixes nightmare being selectable by the wizard

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/angel.dm [hunk 1]
```diff
@@ -5,9 +5,9 @@
 	species_traits = list(EYECOLOR,HAIR,FACEHAIR,LIPS)
 	mutant_bodyparts = list("tail_human", "ears", "wings")
 	default_features = list("mcolor" = "FFF", "tail_human" = "None", "ears" = "None", "wings" = "Angel")
-	use_skintones = 1
+	use_skintones = TRUE
 	no_equip = list(slot_back)
-	blacklisted = 1
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/angel.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species_types/corporate.dm [hunk 2]
```diff
@@ -13,7 +13,7 @@
 	punchstunthreshold = 25
 	attack_verb = "smash"
 	attack_sound = 'sound/weapons/resonator_blast.ogg'
-	blacklisted = 1
-	use_skintones = 0
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/corporate.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species_types/plasmamen.dm [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
 	meat = /obj/item/stack/sheet/mineral/plasma
 	species_traits = list(NOBLOOD,RESISTCOLD,RADIMMUNE,NOTRANSSTING,VIRUSIMMUNE,NOHUNGER)
 	mutantlungs = /obj/item/organ/lungs/plasmaman
 	mutanttongue = /obj/item/organ/tongue/bone/plasmaman
 	mutantliver = /obj/item/organ/liver/plasmaman
 	mutantstomach = /obj/item/organ/stomach/plasmaman
-	dangerous_existence = 1 //So so much
-	blacklisted = 1 //See above
+	dangerous_existence = TRUE //So so much
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/plasmamen.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -34,7 +34,7 @@
 	var/toxic_food = TOXIC
 	var/list/no_equip = list()	// slots the race can't equip stuff to
 	var/nojumpsuit = 0	// this is sorta... weird. it basically lets you equip stuff that usually needs jumpsuits without one, like belts and pockets and ids
-	var/blacklisted = 0 //Flag to exclude from green slime core species.
+	var/blacklisted = FALSE //Flag to exclude from green slime core species or the magic mirror
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species.dm]: Make it a flags var instead

File: code/modules/mob/living/carbon/human/species_types/shadowpeople.dm [hunk 5]
```diff
@@ -28,6 +28,7 @@
 	id = "nightmare"
 	limbs_id = "shadow"
 	burnmod = 1.5
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/shadowpeople.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species_types/shadowpeople.dm [hunk 6]
```diff
@@ -28,6 +28,7 @@
 	id = "nightmare"
 	limbs_id = "shadow"
 	burnmod = 1.5
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
 	blacklisted = TRUE
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/shadowpeople.dm]: You didn't remove the old line

File: code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 7]
```diff
@@ -2,10 +2,10 @@
 	name = "Synth" //inherited from the real species, for health scanners and things
 	id = "synth"
 	say_mod = "beep boops" //inherited from a user's real species
-	sexes = 0
+	sexes = FALSE
 	species_traits = list(NOTRANSSTING,NOBREATH,VIRUSIMMUNE,NODISMEMBER,NOHUNGER) //all of these + whatever we inherit from the real species
 	dangerous_existence = 1
-	blacklisted = 1
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/synths.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species_types/skeletons.dm [hunk 8]
```diff
@@ -3,8 +3,8 @@
 	name = "Spooky Scary Skeleton"
 	id = "skeleton"
 	say_mod = "rattles"
-	blacklisted = 1
-	sexes = 0
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/skeletons.dm]: | SPECIES_BLACKLIST_SLIME_MUT

File: code/modules/mob/living/carbon/human/species_types/zombies.dm [hunk 9]
```diff
@@ -20,6 +20,7 @@
 	armor = 20 // 120 damage to KO a zombie, which kills it
 	speedmod = 2
 	mutanteyes = /obj/item/organ/eyes/night_vision/zombie
+	blacklisted = SPECIES_BLACKLIST_MAGIC_MIRROR
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/zombies.dm]: | SPECIES_BLACKLIST_SLIME_MUT

**Review by Cyberboss — CHANGES_REQUESTED** [review]
You also need to change the checks for the blacklisted var

**Comment by KorPhaeron** [comment]
@Cyberboss 

Should I just put these in with the rest of the species flags then, or move the other blacklist related vars (changeling stings) out of the species flags and into this new list?

**Comment by Cyberboss** [comment]
go with the second

File: code/modules/mob/living/carbon/human/species.dm [hunk 10]
```diff
@@ -56,6 +55,7 @@
 
 	// species flags. these can be found in flags.dm
 	var/list/species_traits = list()
+	var/list/species_blacklist = list()
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species.dm]: why this be a list and not a flag, u only got 4 things yo

Comment (optimumtact) [reply code/modules/mob/living/carbon/human/species.dm]: we paying init overhead for all these lists, every single one on every single species datum

Comment (optimumtact) [reply code/modules/mob/living/carbon/human/species.dm]: you didn't even make a lazy init list shame on you

**Comment by XDTM** [comment]
Add no trans sting and no dna copy to android, no way a ling can get its dna!

**Comment by KorPhaeron** [comment]
Between getting ready for classes and divinity 2 I realistically will not work further on this

**Pull request closed without merging**